src/main/java/de/kittybot/kittybot/database/Database.java

```diff
@@ -0,0 +1,324 @@
+			LOG.error("Error inserting self-assignable roles: " + roles.toString(), e);
+		}
+	}
+
+	public static Map<String, String> getSelfAssignableRoles(String guildId){
+		try{
+			var ctx = getCtx();
+			return ctx.selectFrom(SELF_ASSIGNABLE_ROLES).where(SELF_ASSIGNABLE_ROLES.GUILD_ID.eq(guildId)).fetch()
+					.stream()
+					.collect(Collectors.toMap(sar -> sar.get(SELF_ASSIGNABLE_ROLES.ROLE_ID), sar -> sar.get(SELF_ASSIGNABLE_ROLES.ROLE_ID)));
+		}
+		catch(SQLException e){
+			LOG.error("Error while getting self-assignable roles from guild " + guildId, e);
+		}
+		return null;
+	}
+
+	public static String getAnnouncementChannelId(String guildId){
+		return getProperty(guildId, GUILDS.ANNOUNCEMENT_CHANNEL_ID);
+	}
+
+	public static void setAnnouncementChannelId(String guildId, String channelId){
+		setProperty(guildId, GUILDS.ANNOUNCEMENT_CHANNEL_ID, channelId);
+	}
+
+	public static String getJoinMessage(String guildId){
+		return getProperty(guildId, GUILDS.JOIN_MESSAGES);
+	}
+
+	public static void setJoinMessage(String guildId, String message){
+		setProperty(guildId, GUILDS.JOIN_MESSAGES, message);
+	}
+
+	public static Boolean getJoinMessageEnabled(String guildId){
+		return getProperty(guildId, GUILDS.JOIN_MESSAGES_ENABLED);
+	}
+
+	public static void setJoinMessageEnabled(String guildId, boolean enabled){
+		setProperty(guildId, GUILDS.JOIN_MESSAGES_ENABLED, enabled);
+	}
+
+	public static String getLeaveMessage(String guildId){
+		return getProperty(guildId, GUILDS.LEAVE_MESSAGES);
+	}
+
+	public static void setLeaveMessage(String guildId, String message){
+		setProperty(guildId, GUILDS.LEAVE_MESSAGES, message);
+	}
+
+	public static Boolean getLeaveMessageEnabled(String guildId){
+		return getProperty(guildId, GUILDS.LEAVE_MESSAGES_ENABLED);
+	}
+
+	public static void setLeaveMessageEnabled(String guildId, boolean enabled){
+		setProperty(guildId, GUILDS.LEAVE_MESSAGES_ENABLED, enabled);
+	}
+
+	public static String getBoostMessage(String guildId){
+		return getProperty(guildId, GUILDS.BOOST_MESSAGES);
+	}
+
+	public static void setBoostMessage(String guildId, String message){
+		setProperty(guildId, GUILDS.BOOST_MESSAGES, message);
+	}
+
+	public static Boolean getBoostMessageEnabled(String guildId){
+		return getProperty(guildId, GUILDS.BOOST_MESSAGES_ENABLED);
+	}
+
+	public static void setBoostMessageEnabled(String guildId, boolean enabled){
+		setProperty(guildId, GUILDS.BOOST_MESSAGES_ENABLED, enabled);
+	}
+
+	public static Boolean getNSFWEnabled(String guildId){
+		return getProperty(guildId, GUILDS.NSFW_ENABLED);
+	}
+
+	public static void setNSFWEnabled(String guildId, boolean enabled){
+		setProperty(guildId, GUILDS.NSFW_ENABLED, enabled);
+	}
+
+	public static ReactiveMessage getReactiveMessage(String guildId, String messageId){
+		try{
+			var ctx = getCtx();
+			var reactiveMsg = ctx.selectFrom(REACTIVE_MESSAGES).where(REACTIVE_MESSAGES.MESSAGE_ID.eq(messageId).and(REACTIVE_MESSAGES.GUILD_ID.eq(guildId))).fetchOne();
```

Comment:
is checking the guild id here useful? message ids are unique for each guild. might want to explain?

Reply:
it's not needed sure but at the same time why not I guess
We do this at work too just to be safe